### PR TITLE
Feature/save favourites route

### PR DESCRIPTION
**Pull request by MGS31**
DON"T FORGET NPM I  :P

Finished the structure to allow a user to add a city to their favourites

- [x] Should allow the button to add once a user has searched for a favourite
- [x] Should show the favourite with a unified capitalised first letter

As always let me know if there are any issues with the response time or the code breaks but it was working fine for me now!

### CODE REVIEW

File: leafsmart-back-end/index.js [hunk 1]
```diff
@@ -15,13 +15,15 @@ const db = knex({
 });
 
 const favRoute = require("./routes/favouriteRouter");
+const savFavRoute = require('./routes/saveFavouritesRouter')
```
Comment (b1u3too) [inline leafsmart-back-end/index.js:18]: I'm a little confused with these routes! Can we talk about them as a group? I'm confused about why all of the resource routes (create, read, delete) wouldn't all be in a single router file.

File: leafsmart-front-end/src/components/buttons/AddFavourite.js [hunk 2]
```diff
@@ -1,8 +1,28 @@
...
+  const [geoID, setGeoID] = useState(null)
+  const [displayName, setDisplayName] = useState("")
+  const [cityName, setCityName] = useState("")
+
+  const { saveFavourites } = useAuth();
+
+  useEffect(() => {
+    setGeoID(props.geonameId);
+    setDisplayName(props.cName);
+    setCityName(props.cName);
```
Comment (b1u3too) [inline leafsmart-front-end/src/components/buttons/AddFavourite.js:14]: Can we destructure out these props at the top of the component? Will make understanding what props are expected to be passed in much easier! 
eg. `const { geonameId, cName } = props;`

**Review by b1u3too — CHANGES_REQUESTED** [review]
This is super rad progress! I'd like to talk about the router file structure from the comments before we merge, and:
- Right now a user can "mash that subscribe button" and add a favourite multiple times. Should they be allowed to? 
image

- Can a user remove a favourite, should/will we be able to implement that? 
- How will the user know the button worked? (Active button styling, favourites tray opening to show their add... a notification)... we can talk about these things and work on them in future pull requests!

File: leafsmart-back-end/index.js (hunk 1 above)
Comment (MGS31) [reply leafsmart-back-end/index.js:18]: We can clean up the routes in the back end, right now it's sorted around how the original requests were handled when testing. Understanding they are not super clean this was just getting everything working from backend request.

**Comment by MGS31** [comment]
@b1u3too all good points in the end, a button to delete from Favourites makes sense additionally it would solve the issue with the multiple favourites being added but right now I have a "hack" returning only unique names from the favourites table. for passing into the favourites box. 

A better fix would be to remove the favourites button after a favourite has been added so they can't add multiples of the same location.

A lot of this beyond the delete favourites feels like stretch work so maybe we come back to this once we have all features implemented? 

I think a return of the favourite to the search would be more important in the MVP of the product???  thoughts?

**Review by rachelpr — APPROVED** [review]

**Review by b1u3too — APPROVED** [review]
MAGNIFICENT!

**Pull request merged**